### PR TITLE
Allowing pregenerated files in python build

### PR DESCRIPTION
**Pull request by oidebrett**
modified build_python.sh to take argument to use pregenerated files as per issue #24718 - Build Python fails due to Zap Generation on Raspberry Pi (no ARM based version of zap-cli available). Alos updated matter-repl.md to explain how to use

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: docs/guides/matter-repl.md [hunk 1]
```diff
@@ -25,11 +25,71 @@ You can find source files of the Python CHIP Controller tool in the
...
+    ```
+    scripts/build_python.sh -m platform -i separate
+    ```
+
+    > Note: To get more details about available build configurations, run the
+    > following command: `scripts/build_python.sh --help`
+
+### Building using pregenerated zap files e.g. for Raspberry Pi 4
+
+1. Building for Arm based systems such as the Raspberry Pi 4 should use pregenerated files since there are currently no pre-built zap Arm Linux binaries. To understand about code generation and pregenerating code files see [Code generation](../code_generation.md#Pre-generation). To build and install the Python CHIP controller with pregenerated files use the --pregen_dir (-z) argument:
```
Comment (bzbarsky-apple) [inline docs/guides/matter-repl.md:85]: The docs there don't document which things need to be pregenerated, right?  That is, _which specific args_ to run pregen with.

That probably needs to be documented here.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@oidebrett Restyle failures?

**Comment by oidebrett** [comment]
> @oidebrett Restyle failures?

Hi, sorry about that. This is my first PR to this repo. I have made the restyling changes as requested and added some extra information on what is being pregenerated. 

Could I just check if I am doing this PR push correctly?

I am following the Contributing guide: https://github.com/oidebrett/connectedhomeip/blob/master/CONTRIBUTING.md#submitting-a-pull-request

When I get to final step to "Push and Test" I get an error:
git push origin pregenerated_python_build
To https://github.com/oidebrett/connectedhomeip.git
 ! [rejected]            pregenerated_python_build -> pregenerated_python_build (non-fast-forward)
error: failed to push some refs to 'https://github.com/oidebrett/connectedhomeip.git'
hint: Updates were rejected because the tip of your current branch is behind

When I do a git status the only files that are modified in this branch compared to my master repo branches are the third party submodule files that I guess were brought in when I did a git submodule update --init. e.g. third_party/infineon/psoc6/psoc6_sdk/libs/serial-flash

Should I use -f to force this? Whats the best practice

Thanks for the advice. I enjoy contributing to this amazing project!!

**Comment by bzbarsky-apple** [comment]
> Hi, sorry about that.

No need to apologize!

> Should I use -f to force this? Whats the best practice

It depends on how you applied your changes.

If you created a new changeset for the changes, then I'm surprised you are getting the errors you are getting, and we should figure out what's actually going on there.  Please let me know if that's the case.

If you amended your existing changeset, or rebased things, then yes just `-f` to force.

**Comment by bzbarsky-apple** [comment]
@oidebrett Still restyle failures.  See https://restyled.io/gh/project-chip/repos/connectedhomeip/jobs/2798166/patch for the patch the restyler is suggesting.

**Comment by bzbarsky-apple** [comment]
@oidebrett Thank you for the PR!

**Pull request merged**